nerfstudio/model_components/losses.py

```diff
@@ -456,47 +457,6 @@ def __init__(self, alpha: float = 0.5, scales: int = 4, reduction_type: Literal[
 
         self.__prediction_ssi = None
 
-    @classmethod
-    def compute_scale_and_shift(
-        cls, prediction: TensorType[1, 32, "mult"], target: TensorType[1, 32, "mult"], mask: TensorType[1, 32, "mult"]
-    ):
-        """
-        More info here: https://arxiv.org/pdf/2206.00665.pdf supplementary section A2 Depth Consistency Loss
-        This function computes scale/shift required to normalizes predicted depth map,
-        to allow for using normalized depth maps as input from monocular depth estimation networks.
-        These networks are trained such that they predict normalized depth maps.
-
-        Solves for scale/shift using a least squares approach with a closed form solution:
-        Based on:
-        https://github.com/autonomousvision/monosdf/blob/d9619e948bf3d85c6adec1a643f679e2e8e84d4b/code/model/loss.py#L7
-        Args:
-            prediction: predicted depth map
-            target: ground truth depth map
-            mask: mask of valid pixels
-        Returns:
-            scale and shift for depth prediction
-        """
-        # system matrix: A = [[a_00, a_01], [a_10, a_11]]
-        a_00 = torch.sum(mask * prediction * prediction, (1, 2))
-        a_01 = torch.sum(mask * prediction, (1, 2))
-        a_11 = torch.sum(mask, (1, 2))
-
-        # right hand side: b = [b_0, b_1]
-        b_0 = torch.sum(mask * prediction * target, (1, 2))
-        b_1 = torch.sum(mask * target, (1, 2))
-
-        # solution: x = A^-1 . b = [[a_11, -a_01], [-a_10, a_00]] / (a_00 * a_11 - a_01 * a_10) . b
-        scale = torch.zeros_like(b_0)
-        shift = torch.zeros_like(b_1)
-
-        det = a_00 * a_11 - a_01 * a_01
-        valid = det.nonzero()
-
-        scale[valid] = (a_11[valid] * b_0[valid] - a_01[valid] * b_1[valid]) / det[valid]
-        shift[valid] = (-a_01[valid] * b_0[valid] + a_00[valid] * b_1[valid]) / det[valid]
-
-        return scale, shift
-
     def forward(
         self, prediction: TensorType[1, 32, "mult"], target: TensorType[1, 32, "mult"], mask: TensorType[1, 32, "mult"]
```

Comment:
Might be confused here, but why is the argument shape here `TensorType[1, 32, "mult"]`? Does it not depend on the batch size?

Reply:
in the original SDFStudio implementation they always reshape the depth map to have a size of [1, 32, -1] which I did [here](https://github.com/nerfstudio-project/nerfstudio/blob/688178e272ebcb7e0d63bdfc444cbda32490ba4b/nerfstudio/models/base_surface_model.py#L315). I don't know as to why this was done. I could also rename "mult" and just use -1 (I haven't used torchtyping before some I'm still figuring out how its used). Let me know what you'd like me to do with it!

Reply:
Gotcha. For the sake of the consistency in this repo it might make sense to change the forward pass of this loss to something of shape [batch size, ...] and get rid of the reshpae in the field code instead, even though they did this in the SDF implementation. I can't imagine it changed the computed loss results at all.

Reply:
Yep, no worries, as I'm definitely still not an expert in neural rendering or SDF-related things I'm trying to learn as much as I can to hopefully make good distinctions on this. I'm heavily leaning on the implementation that SDFStudio made as I go through the papers and try to get a better understanding of how things work and why they work (as well as the recommendations you all provide for code structure and implementation).